### ISSUE TITLE
Support of Struts 7

### ISSUE THREAD

**Issue by afattahi54** [struts-community-plugins/struts2-jquery#470]
I recently find that the next version of struts will be [Struts 7](https://cwiki.apache.org/confluence/display/WW/Version+Notes+7.0.0-M1)  which will migrate to JakartaEE

I wonder if there is any plane for this upgrade 

Thanks

**Comment by azhigmont** [comment]
Any plans for this upgrade?

**Comment by lukaszlenart** [comment]
There is a conflict between Freemarker versions, it has been resolved in Struts 7 branch yet no new Milestone version to allow start working on upgrade.

**Comment by azhigmont** [comment]
> There is a conflict between Freemarker versions, it has been resolved in Struts 7 branch yet no new Milestone version to allow start working on upgrade.

The milestone version of Struts 7.x is already available https://lists.apache.org/thread/vns833ospsvvw5p2qzh34po4sdlfy155

**Comment by lukaszlenart** [comment]
Yes, there is already the M7 as well but it's missing this fix https://github.com/apache/struts/pull/970

**Comment by Foxfire1st** [comment]
> Yes, there is already the M7 as well but it's missing this fix [apache/struts#970](https://github.com/apache/struts/pull/970)

I ran now into this issue too when I migrated to 7.x.x from 2.5. I have Struts 7 and Spring Boot 3.1 running now. And this plugin was causing

**Comment by dja557** [comment]
Yes, there is also an issue here with the integration of Spring Boot 3.3.3, struts 7.0.0-M9 and`struts2-bootstrap-plugin` version `4.0.3`:

```
java.lang.AbstractMethodError: Receiver class com.jgeppert.struts2.bootstrap.views.BootstrapTagLibrary does not define or inherit an implementation of the resolved method 'abstract java.lang.Object getModels(com.opensymphony.xwork2.util.ValueStack, jakarta.servlet.http.HttpServletRequest, jakarta.servlet.http.HttpServletResponse)' of interface org.apache.struts2.views.TagLibraryModelProvider.
```

Any update here?

**Comment by rrangelo** [comment]
_For English speakers:_

Hello, I wanted to know if there has been any update on this topic, as I am currently working on the migration of some projects that use the following libraries:

com.jgeppert.struts2.jquery
com.jgeppert.struts2.jquery
com.jgeppert.struts2.jquery

Originally, the code is in Java 1.6 and Struts 2.3.1.1, and we are migrating to Java 17 and Struts 6.6.0 to deploy on WildFly 27.0.1.Final. In these new versions, they use 'jakarta.\*' instead of 'javax.\*', so it is essential for our team to migrate the libraries to versions that support 'jakarta.\*'.

I might be reaching out in the wrong repository, and there could be another one where this update has already been done. If so, could you please share it?

---

_Para hispanohablantes:_

Hola, quería saber si tenían alguna actualización sobre este tema, ya que actualmente me encuentro trabajando en la migración de unos proyectos que usan estas librerías:

com.jgeppert.struts2.jquery:struts2-jquery-plugin
com.jgeppert.struts2.jquery:struts2-jquery-richtext-plugin
com.jgeppert.struts2.jquery:struts2-jquery-grid-plugin

Originalmente, el código está en Java 1.6 y Struts 2.3.1.1, y estamos migrando, inicialmente, a Java 17 y Struts 6.6.0 para desplegar en WildFly 27.0.1.Final. En estas nuevas versiones usan "jakarta.\*" y no "javax.\*", por lo que nos es imperativo en el equipo migrar las librerías a versiones que soporten "jakarta.\*".

Tal vez estoy consultando en el repositorio equivocado y existe alguno donde esta actualización ya haya sido hecha previamente, si es así, me la podrían compartir?

**Comment by gregh3269** [comment]
Struts 6.6.0 is 'javax.*' not 'jakarta.*'.  You will need to use v7 for  'jakarta.*' which is soon to be released.

**Comment by dja557** [comment]
@gregh3269 - Is there a release date set for v7?

**Comment by gregh3269** [comment]
Not sure of the exact date but in November hopefully.  You can try M9 which is quite stable.

https://struts.apache.org/announce-2024#a20240721

**Comment by dja557** [comment]
Thanks! Where do we log issues with M9 if we find any?

**Comment by gregh3269** [comment]
Help testing would be good.  I see them on the dev list. But not sure if that is the correct place @lukaszlenart?

https://struts.apache.org/dev-mail.html

**Comment by lukaszlenart** [comment]
Dev list is preferable

**Comment by mcanzerini** [comment]
Any plans for this upgrade?

**Comment by lukaszlenart** [comment]
Rather not quickly, I can try looking into it once first release of Struts 7 is out.

**Comment by lukaszlenart** [comment]
PR #597 is ready for review :)
